extensions/wikia/CommunityHeader/tests/NavigationTest.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+namespace CommunityHeader;
+
+use Wikia\Util\GlobalStateWrapper;
+
+class NavigationTest extends \WikiaBaseTest {
+	/**
+	 * @dataProvider exploreItemsProviders
+	 *
+	 * @param $globals
+	 * @param $expectedExploreItems
+	 */
+	public function testExploreItems( $globals, $expectedExploreItems ) {
+		$globals = new GlobalStateWrapper( [
+			'wgEnableCommunityPageExt' => $globals['wgEnableCommunityPageExt'],
+			'wgEnableDiscussions' => $globals['wgEnableDiscussions'],
+			'wgEnableForumExt' => $globals['wgEnableForumExt'],
+		] );
+
+		$result = $globals->wrap( function () {
+			return new Navigation();
+		} );
+
+		// used `array_values` to reset keys of array
+		$this->assertEquals( $expectedExploreItems, array_values( $result->exploreItems ) );
+	}
+
+	/**
+	 * @dataProvider discussLinkProviders
+	 *
+	 * @param $globals
+	 * @param $expected
+	 */
+	public function testDiscussLink( $globals, $expected ) {
+		$globals = new GlobalStateWrapper( [
+			'wgEnableDiscussions' => $globals['wgEnableDiscussions'],
+			'wgEnableForumExt' => $globals['wgEnableForumExt'],
+		] );
+
+		$result = $globals->wrap( function () {
+			return new Navigation();
+		} );
+
+		$this->assertEquals( $expected, $result->discussLink );
+	}
+
+	private function prepareExploreItems( $raw ) {
+		return array_map( function ( $rawItem ) {
+			return new Link(
+				new Label( $rawItem['label']['key'], $rawItem['label']['type'] ),
+				$rawItem['href'],
+				$rawItem['tracking']
+			);
+		}, $raw );
+	}
+
+	public function exploreItemsProviders() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "exploreItemsProviders" has 257 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)